modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/ControlAcceleratorSupport.java

```diff
@@ -272,7 +286,8 @@ public static void removeAcceleratorsFromScene(List<? extends MenuItem> items, S
                 final Map<KeyCombination, Runnable> accelerators = scene.getAccelerators();
                 accelerators.remove(menuitem.getAccelerator());
 
-                ChangeListener<KeyCombination> listener = changeListenerMap.get(menuitem);
+                WeakReference<ChangeListener<KeyCombination>> listenerW = changeListenerMap.get(menuitem);
+                ChangeListener<KeyCombination> listener = listenerW == null ? null : listenerW.get();
                 if (listener != null) {
```

Comment:
This will fail to remove a weak reference to a listener that has been collected. It is a `WeakHashMap`, so the `WeakReference` to the listener will be reclaimed as soon as the `menuItem` is no longer reachable, but if you used the same logic as you did for the scene listener (e.g., on lines 254-261), it would be removed sooner. I don't know whether it matters in this case.

Reply:
Yes, you are right. I now use the same logic in all 3 places.
I guess it doesn't matter because the key always references the listener - but the code is now more straight line.